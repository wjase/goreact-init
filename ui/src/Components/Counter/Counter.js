import React from 'react';

export const Counter = () => {

    const [count, setCount] = React.useState(0);

    return (
        <div>
            <div>
                <span id='counttext'>{count}</span>
            </div>
      <button id='incr_btn' type="button" onClick={() => setCount(count + 1)}>
        Increment
      </button>

      <button id='decr_btn' type="button" onClick={() => setCount(count - 1)}>
        Decrement
      </button>


        </div>
    );


};

export default Counter;