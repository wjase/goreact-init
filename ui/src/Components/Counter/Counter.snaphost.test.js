import React from 'react';
import { mount } from 'enzyme'; 

import Counter from './Counter.js'; 

describe('Counter', () => {
    it('should render correctly ', () => {
        const component = mount(<Counter />);
        expect(component).toMatchSnapshot();
    });
});