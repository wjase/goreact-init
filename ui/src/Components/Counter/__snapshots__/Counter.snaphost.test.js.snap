// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render correctly  1`] = `
<Counter>
  <div>
    <div>
      <span
        id="counttext"
      >
        0
      </span>
    </div>
    <button
      id="incr_btn"
      onClick={[Function]}
      type="button"
    >
      Increment
    </button>
    <button
      id="decr_btn"
      onClick={[Function]}
      type="button"
    >
      Decrement
    </button>
  </div>
</Counter>
`;
